channels/api.py

```diff
@@ -432,9 +432,10 @@ def add_moderator(self, moderator_name, channel_name):
             user = User.objects.get(username=moderator_name)
         except User.DoesNotExist:
             raise NotFound("User {} does not exist".format(moderator_name))
-
-        self.get_channel(channel_name).moderator.add(user)
-        Api(user).accept_invite(channel_name)
+        channel = self.get_channel(channel_name)
+        if moderator_name not in channel.moderator():
+            channel.moderator.add(user)
+            Api(user).accept_invite(channel_name)
```

Comment:
It looks like an error is raised if there is no invite already. Can you wrap this in a `try: except APIException:` block and log the exception if one happens?